lib/puppet/gettext/config.rb

```diff
@@ -30,12 +47,43 @@ def self.translation_mode(conf_path)
     end
   end
 
+  # Prevent future gettext initializations
+  def self.disable_gettext
+    @gettext_disabled = true
+  end
+
   # Attempt to initialize the gettext-setup gem
   # @param path [String] to gettext config file
   # @param file_format [Symbol] translation file format to use, either :po or :mo
   # @return true if initialization succeeded, false otherwise
   def self.initialize(conf_file_location, file_format)
-    # Bypass gettext until we can resolve a performance regression related to it, PUP-8009.
-    return false
+    return if @gettext_disabled
```

Comment:
Be explicit about the `return false` part?

Reply:
Oh yeah, `return false if @gettext_disabled` would be better.